lib/Interpreter/DeclUnloader.cpp

```diff
@@ -711,10 +736,49 @@ bool DeclUnloader::VisitRedeclarable(clang::Redeclarable<T>* R, DeclContext* DC)
 
   bool DeclUnloader::VisitNamespaceDecl(NamespaceDecl* NSD) {
     // NamespaceDecl: NamedDecl, DeclContext, Redeclarable
-    bool Successful = VisitDeclContext(NSD);
-    Successful &= VisitRedeclarable(NSD, NSD->getDeclContext());
+
+    // When unloading a NamedDecl within an inline namespace within std the
+    // NamedDecl needs to be removed from the parent std as well.
+    // Not sure why, but test/CodeUnloading/NameSpaces.C demonstrates
```

Comment:
I am confused.

```
// T.h 
 namespace A {
   inline namespace __1 { int f; }
 }
// B.cpp
namespace A{}

#include "T.h"

```

Clang seems to put the lookup information in the parent non-inline enclosing namespace:

```
clang -std=c++11 -fsyntax-only -Xclang -ast-dump-lookups -Xclang -ast-dump-filter -Xclang A B.cpp 
Dumping A:
StoredDeclsMap Namespace 0x7f8253837e50 'A'
|-DeclarationName 'f'
| `-Var 0x7f8253837fa0 'f' 'int'
`-DeclarationName '__1'
  `-Namespace 0x7f8253837f20 '__1'

Dumping A:
Lookup map is in primary DeclContext 0x7f8253837e78
```
It seems that unloading T.h would leave the lookup table of A broken.